src/dlgPackageExporter.cpp

```diff
@@ -268,64 +317,230 @@ void dlgPackageExporter::slot_export_package()
+                // two commented out bits of code can be restored
+                if (fileEntries.contains(QStringLiteral("config.lua"))) {
+                    if (!writeFileToZip(QStringLiteral("config.lua"), fileEntries.value(QStringLiteral("config.lua")), archive)) {
+                        /* isOk = false; */
+                    } else {
+                        fileEntries.remove(QStringLiteral("config.lua"));
+                    }
+                }/* else {
+                    displayResultMessage(tr("Required \"config.lua\" file not found to include in module (archive) file, did you remove or rename it?"), false);
+                    zip_close(archive);
+                    isOk = false;
+                }*/
+            }
+
+            QString xmlFileName = mPackageName;
+            xmlFileName.append(QLatin1String(".xml"));
+            if (isOk) {
+                QMapIterator<QString, QString> itFileName(fileEntries);
+                while (itFileName.hasNext() && isOk) {
+                    itFileName.next();
+                    if (itFileName.key() == xmlFileName) {
+                        continue;
+                    }
+
+                    if (!writeFileToZip(itFileName.key(), itFileName.value(), archive)) {
+                        zip_close(archive);
+                        isOk = false;
+                        break;
+                    }
+                }
+            }
+
+            if (isOk) {
+                if (fileEntries.contains(xmlFileName) && fileEntries.value(xmlFileName) == mXmlPathFileName) {
+                    if (!writeFileToZip(xmlFileName, mXmlPathFileName, archive)) {
+                        isOk = false;
+                    }
+
+                    // If successful will get to HERE...
+
+                } else {
+                    displayResultMessage(tr("Required \"%1\" file that contains the mudlet items chosen for the "
+                                            "module was not found in the staging area from where it would be include in the "
+                                            "module (archive) file - this suggests there is a problem with the "
+                                            "permissions for the \"%2\" directory!")
+                                            .arg(mXmlPathFileName, QDir(mStagingDirName).canonicalPath()), false);
+                    isOk = false;
+                }
+            }
+
+            if (isOk) {
+                // THIS is the point that the archive gets created from the
+                // source materials - it may take a short while!
+                // If it fails to write out the new file 'archive' is left
+                // unchanged (and we can still access it to get the error
+                // details):
+                ze = zip_close(archive);
+                if (ze) {
+                    displayResultMessage(tr("Failed to populate and then close module (archive) file, error is: \"%1\".")
```

Comment:
`populate and then close` is very tech-focused and correct, but more understandable would be `write`. Also, package?

Reply:
I'd want to review the other texts to check whether the word "write" is used in them - I would rather like to keep the error messages as distinct as possible so that if a user moans about something going wrong for them we can work out precisely where something went wrong.

Reply:
I've been through the messages and converted *module* to **(.zip format archive) package file** - which seems to more precisely define the item - does that make better sense?